### PR TITLE
Minor doc update for queue.buffering.max.*

### PR DESCRIPTION
**Pull request by briot**
The producer queue is shared for all topics and partitions (we do
 not have one queue per toppar).

### CODE REVIEW

**Comment by edenhill** [comment]
The count and size are per producer, not per partition:
https://github.com/edenhill/librdkafka/blob/master/src/rdkafka_int.h#L296

So your doc change is fine, but it needs to go in rdkafka_conf.c, then rebuild and commit both rdkafka_conf.c and CONFIGURATION.md, the latter being auto generated from the former.
Make sure to only commit lines to CONFIGURATION.md that you actually intended since its output depends on build options.

**Comment by briot** [comment]
I had forgotten this was auto-generated, will amend the commit thanks.

After some more thoughts on the topic, it had seemed like the producer queue limits (count and size) might be per broker. If a producer sends to two brokers (because messages are sent to different topics/partition), are we still sharing the same queue ?

On the one hand, it makes it easier to limit memory usage by the producer, but it also makes it somewhat harder to control the batches in particular to get better compression

**Comment by edenhill** [comment]
These limits are global per rd_kafka_t instance, not per partition nor broker.

File: src/rdkafka_conf.c [hunk 1]
```diff
@@ -917,12 +917,15 @@ static const struct rd_kafka_property rd_kafka_properties[] = {
         { _RK_GLOBAL|_RK_PRODUCER|_RK_HIGH, "queue.buffering.max.messages",
           _RK_C_INT,
 	  _RK(queue_buffering_max_msgs),
-	  "Maximum number of messages allowed on the producer queue.",
+	  "Maximum number of messages allowed on the producer queue. "
+	  "This queue is shared by all topics and partitions.",
+
```
Comment (edenhill) [inline src/rdkafka_conf.c]: remove blank line

**Comment by edenhill** [comment]
Thank you!

**Pull request merged**